hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -182,38 +184,55 @@ def __init__(
         s3_bucket: t.Any,  # Not typable?
         s3_te_data_folder: str,
         data_store_table: str,
+        supported_signal_types: t.List[SignalType],
     ) -> None:
         super().__init__(privacy_group)
         self.app_id = app_id
         self._cached_state: t.Optional[t.Dict] = None
         self.s3_bucket = s3_bucket
         self.s3_te_data_folder = s3_te_data_folder
         self.data_store_table = data_store_table
+        self.supported_signal_types = supported_signal_types
+
+    @functools.lru_cache(maxsize=None)
+    def _get_supported_indicator_types(self):
+        """
+        For supported self.signal_types, get their corresponding indicator_types.
+        """
+        indicator_types = [
+            "HASH_VIDEO_MD5"
+        ]  # hardcoding this because md5 currently points to HASH_MD5
+
+        for signal_type in self.supported_signal_types:
+            indicator_type = getattr(signal_type, "INDICATOR_TYPE", None)
+            if indicator_type:
+                indicator_types.append(indicator_type)
+            else:
+                warnings.warn(
+                    f"SignalType: {signal_type} does not provide an indicator type."
+                )
```

Comment:
nit: If this code is reached its like the result of a bug right? Feels like it should be a error/exception rather than warn

Reply:
It is inside a loop, so I'd rather warn and process the next signal_type if one exists.